gallery/src/components/NxProgressBar/NxProgressBarExample.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { NxProgressBar } from '@sonatype/react-shared-components';
+
+export default function NxProgressBarExample() {
+  return (
+    <>
+      <div>
+        <NxProgressBar max={50}
```

Comment:
According to https://www.w3.org/TR/wai-aria-1.2/#progressbar all progressbars must have accessible names.  This one doesn't, and ChromeVox skips over it without reading anything or even stopping on it.

Reply:
> According to https://www.w3.org/TR/wai-aria-1.2/#progressbar all progressbars must have accessible names. This one doesn't, and ChromeVox skips over it without reading anything or even stopping on it.

I made label required here: 1794de009617bd465405ca7af6f8ec11824f0f72
When `variant` is "small", "inline", or `inlineCounter` is true (the label is hidden), the label is assigned to aria-label on the progress element. I also changed the label tag to span when variant is "inline".

Not entirely sure if this is a good solution 🤔 let me know your thoughts.

Reply:
> I made label required here: https://github.com/sonatype/sonatype-react-shared-components/commit/1794de009617bd465405ca7af6f8ec11824f0f72
> When variant is "small", "inline", or inlineCounter is true (the label is hidden), the label is assigned to aria-label on the progress element. I also changed the label tag to span when variant is "inline".
>
> Not entirely sure if this is a good solution thinking let me know your thoughts.


That seems good to me.  The only case I'm not sure about is if the progress bar is next to some text or something that makes it redundant from a screenreader perspective.  In that case what I would have thought would make sense would be to include no label on the progress bar and to give it a role of `none` (or `presentation` which is the same thing).  However, looking at https://w3c.github.io/html-aria/#el-progress it doesn't seem like that role is allowed on this element which surprises me.  I thought it was allowed on every element.

Reply:
> In that case what I would have thought would make sense would be to include no label on the progress bar and to give it a role of none (or presentation which is the same thing). However, looking at https://w3c.github.io/html-aria/#el-progress it doesn't seem like that role is allowed on this element which surprises me. I thought it was allowed on every element.

Looking at https://www.w3.org/TR/wai-aria/#aria-hidden , it seems like maybe `aria-hidden` is the correct way to accomplish that

Reply:
> > In that case what I would have thought would make sense would be to include no label on the progress bar and to give it a role of none (or presentation which is the same thing). However, looking at https://w3c.github.io/html-aria/#el-progress it doesn't seem like that role is allowed on this element which surprises me. I thought it was allowed on every element.
> 
> Looking at https://www.w3.org/TR/wai-aria/#aria-hidden , it seems like maybe `aria-hidden` is the correct way to accomplish that

Thanks! Made a change here so that aria-hidden is set to true when progress is accompanied by text: a32c4f9f05950afecb13c94e71382d9c6413f817

Additionally, I made `labelError` to be required when `hasError` is true: 6e94da26a79a5bb93d1197df875142b3456e44ac

Reply:
Updated the description here: 2a4f6adc7ec8ee17bb5293d17c89b853248a9451

Reply:
> Thanks! Made a change here so that aria-hidden is set to true when progress is accompanied by text: https://github.com/sonatype/sonatype-react-shared-components/commit/a32c4f9f05950afecb13c94e71382d9c6413f817

That's not what I was suggesting.  What I was suggesting is that `label` not be a required prop, because it is conceivable that an inline progress bar might be used adjacent to some other text or something which makes the label redundant.  In that scenario, the progress component should maybe not be labelled and be hidden from the screenreader.  Maybe we just shouldn't worry about it though.  It's an edge case, it's not the end of the world if we don't do it (it'll just read some redundant content), and I'm not certain the solution we have in mind is standards complaint - does a hidden element get out of needing a label?

Reply:
> That's not what I was suggesting. What I was suggesting is that label not be a required prop, because it is conceivable that an inline progress bar might be used adjacent to some other text or something which makes the label redundant. In that scenario, the progress component should maybe not be labelled and be hidden from the screenreader. Maybe we just shouldn't worry about it though. It's an edge case, it's not the end of the world if we don't do it (it'll just read some redundant content), and I'm not certain the solution we have in mind is standards complaint - does a hidden element get out of needing a label?

Going back to what you had in https://github.com/sonatype/sonatype-react-shared-components/commit/1794de009617bd465405ca7af6f8ec11824f0f72 seems adequate to me

Reply:
> > That's not what I was suggesting. What I was suggesting is that label not be a required prop, because it is conceivable that an inline progress bar might be used adjacent to some other text or something which makes the label redundant. In that scenario, the progress component should maybe not be labelled and be hidden from the screenreader. Maybe we just shouldn't worry about it though. It's an edge case, it's not the end of the world if we don't do it (it'll just read some redundant content), and I'm not certain the solution we have in mind is standards complaint - does a hidden element get out of needing a label?
> 
> Going back to what you had in [1794de0](https://github.com/sonatype/sonatype-react-shared-components/commit/1794de009617bd465405ca7af6f8ec11824f0f72) seems adequate to me

Reverted here: ec7ea78d46a3723b218f42de4188989fb987e028